### PR TITLE
better handle and report errors from unresponsive qry_buffer gen_server

### PR DESCRIPTION
**Pull request by hmmr**
Another attempt to fix `ts_simple_query_buffers_SUITE`.

Also, consume and drop messages riak_kv_qry_N workers are not supposed to handle.

### CODE REVIEW

**Comment by macintux** [comment]
+1

File: src/riak_kv_qry_worker.erl [hunk 1]
```diff
@@ -136,6 +136,10 @@ handle_info({{SubQId, QId}, {error, Reason} = Error},
 handle_info({{_SubQId, QId1}, _}, State = #state{qid = QId2}) when QId1 =/= QId2 ->
     %% catches late results or errors such getting results for invalid QIds.
     lager:debug("Bad query id ~p (expected ~p)", [QId1, QId2]),
+    {noreply, State};
+
+handle_info(_Unexpected, State) ->
+    lager:info("Not handling unexpected message \"~p\" in state ~p", [_Unexpected, State]),
```
Comment (andytill) [inline src/riak_kv_qry_worker.erl:142]: Formatting the unexpected message could result in a very large log message, for this gen_server in particular that handles subquery result chunks with an unbounded number of rows.

Comment (macintux) [reply src/riak_kv_qry_worker.erl:142]: `lager` is designed to truncate large messages.

Comment (hmmr) [reply src/riak_kv_qry_worker.erl:142]: lager has safety checks in place covering this, I hear.

**Comment by andytill** [comment]
+1

**Pull request merged**